README.md

```diff
@@ -1,3 +1,134 @@
+
+For example, being able to handle SSO, JWT, or other methods of authorizing event sources are in our interest to support, as well as the ability to reach out and pull down data from Rest API's using those methods of authorization. With the introduction of this framework we should be able to have a repeatable CI/CD pipeline, and be able to add to the library many functions to handle various types of connectivity and data retrieval without much additional work.
+___
+## Framework Components:
+
+__Cloudformation:__ Cloudformation will create the CI/CD components utilizing:
+- __CodePipeline__
+- __CodeCommit__
+- __CodeBuild__
+- __s3__
+- __CloudWatch__
+- __GitHub__.
+
+The intended behavior is to pull the framework from GitHub, and the custom configuration for a specific event source from its CodeCommit repository. The Cloudformation template has the potential to be separately deployed for various event sources, while the only difference will be the event source's CodeCommit repository that contains the custom configuration data for that event source. This allows us to maintain a generic framework that is event source agnostic and relieved of sensitive data. This structure presents the ability to maintain secret data, and preserve it's confidentiality.
+
+__Serverless:__ The Serverless framework is utilized to deploy the actual application that will handle the event source data. This framework creates the following resources:
+ - A custom authorizer for webhook based APIs that use a header token.
+ - An API Gateway for receiving post events using the authorizer.
+ - Custom configuration variables that are populated by the `buildspec.yml` located in each event source configuration's CodeCommit repository.
+ - A lambda event handler that will validate to some degree the event to be processed by MozDef. At some point we may extend functionality here to fully process the event by another lambda.
+ - An SQS queue to receive the events that have been handled by our lambda handler.
+ - A Dead Letter Queue (DLQ) to receive events that fail to be handled, so that they can be reprocessed once a fix is introduced. This is future functionality that has not been implemented yet.
+
+We plan to include testing modules in this framework as well.
+
+To deploy this framework you will need the following established:
+
+1. You'll need to design your parameter store pathing, and add your tokens or other data to be used with the framework. We used the following structure:
+
+`/<Project_name>/<event_source_name>/<environemnt>/<authorizer_token_var>/`
+
+This allows us to keep track of the various event sources, what environment they are used in, and keep the same variables across all event sources that will contain different values.
+
+2. An AWS Account with the ability to use all of the AWS services previously mentioned.
```

Comment:
I wonder if we should elaborate here a bit more, such as: "For convenience, the build pipeline can be created using the CloudFormation template with an AWS account with administrative rights. This should be OK, as both Cloudformation and Serverless framework create dedicated IAM roles associated with the resources. You can fine tune permissions for these roles the Cloudformation template and/or serverless configuration file (`serverless.yml`)."

This reads a bit long, but something along these lines. Or maybe it's too much detail?

Reply:
You are spot on, I wanted to mention the IAM stuff, but wasn't sure how to word it, this makes sense.

Reply:
I think I'll add this to the bottom of the Readme under the RUN IT section.

Reply:
Done, please review!